packages/paste-core/primitives/box/src/index.tsx

```diff
@@ -10,216 +10,38 @@ import {
-  },
-  borderBottomColor: {
-    property: 'borderBottomColor',
-    scale: 'borderColors',
-  },
-  borderLeftColor: {
-    property: 'borderLeftColor',
-    scale: 'borderColors',
-  },
-  borderRightColor: {
-    property: 'borderRightColor',
-    scale: 'borderColors',
-  },
-  borderTopColor: {
-    property: 'borderTopColor',
-    scale: 'borderColors',
-  },
-  content: true,
-  cursor: true,
-  appearance: true,
-  transition: true,
-  transform: true,
-  animation: true,
-  transformOrigin: true,
-  visibility: true,
-  userSelect: true,
-  pointerEvents: true,
-  boxSizing: true,
-  resize: true,
-  listStyleType: true,
-  listStylePosition: true,
-  listStyleImage: true,
-  objectFit: true,
-  objectPosition: true,
-  outline: true,
-  float: true,
-  willChange: true,
-  clip: true,
-  backgroundAttachment: true,
-  textAlign: true,
-  textTransform: true,
-  textDecoration: true,
-  textOverflow: true,
-  whiteSpace: true,
-  wordBreak: true,
-  wordWrap: true,
-  overflowWrap: true,
-  opacity: true,
-  borderCollapse: true,
-  borderSpacing: true,
-  tableLayout: true,
-  fontVariantNumeric: true,
-});
-
-const getPseudoStyles = (props: BoxProps): unknown => {
-  const pseudoProps = Object.keys(props).filter((propName) => propName.startsWith('_')) as Array<
-    keyof typeof PseudoPropStyles
-  >;
+import type {BoxProps, StyledBoxProps} from './types';
+import {getPseudoStyles, PasteStyleProps, getCustomElementStyles} from './StyleFunctions';
 
-  if (pseudoProps.length === 0) {
-    return {};
-  }
-
-  const pseudoStyles: {[key: string]: any} = {};
-  pseudoProps.forEach((pseudoProp) => {
-    if (PseudoPropStyles[pseudoProp] != null) {
-      pseudoStyles[PseudoPropStyles[pseudoProp]] = props[pseudoProp];
-    }
-  });
-
-  return css(pseudoStyles);
-};
-
-// @ts-ignore
-export const Box = styled.div(
+// @ts-ignore can't work out how to stop the styled div color prop from emotion clashing with our color style prop in BoxProps
+export const StyledBox = styled.div<StyledBoxProps>(
   {
     boxSizing: 'border-box',
   },
-  compose(space, layout, flexbox, background, border, boxShadow, position, typography, extraConfig),
-  getPseudoStyles
-  // we do this because the default typings of emotion styled
-  // means Text gets typed as a span, and can't be extended
-  // correctly to utilise the as prop. The new HTML element attributes
-  // always clash with the span html attributes. To override this,
-  // we retype as a basic functional component which is easy to extend
-) as React.FC<BoxProps>;
+  compose(space, layout, flexbox, background, border, boxShadow, position, typography, PasteStyleProps),
+  getPseudoStyles,
+  getCustomElementStyles
+) as StyledComponent<
+  Omit<React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>, 'color'>,
+  BoxProps,
+  Record<string, unknown>
+>;
+
+const Box = React.forwardRef<HTMLElement, BoxProps>(({children, element = 'BOX', ...props}, ref) => {
```

Comment:
I'm a little curious about your rationale for these `element` props to be all caps. Is it because these are case sensitive and we want to avoid mis-typing things? Or is it rather because we want these to read as constants?

Reply:
Going with case insensitive but making sure you can't have duplicates of the same name in different casings, just in case you get weird cross platform behaviors.